### PR TITLE
Add section about tracestate

### PR DESCRIPTION
**Pull request by felixbarny**

### CODE REVIEW

File: docs/agents/sampling.md [hunk 1]
```diff
@@ -51,14 +53,8 @@ trace-flags: https://www.w3.org/TR/trace-context/#sampled-flag
...
 the sampling weight to ensure a consistent weight is applied to all events in the trace.
-This is achieved by adding an "elastic" key to `tracestate` when calculating the
-sampling weight, and propagating and parsing `tracestate` in all agents.
-
-For our own "elastic" `tracestate` entry we will introduce a key:value formatted list
-of attributes. Initially we will support a single attribute, "w", which holds the
-sample weight. The general `tracestate` format is:
-
-    tracestate: elastic=key:value;key:value...,othervendor=<opaque>
+This is achieved by adding a `w` attribute to our [`elastic` `tracestate` key](agent-development.md#tracestate) when calculating the
```
Comment (axw) [inline docs/agents/sampling.md]:
```suggestion
This is achieved by adding a `w` attribute to our [`elastic` `tracestate` key](distributed-tracing.md#tracestate) when calculating the
```

File: docs/agents/distributed-tracing.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+- The `tracestate` field may contain a maximum of 32 entries.
+  An entry consists of a vendor key, and an opaque vendor value.
+- Vendor keys (such as `elastic`) have a maximum size of 256 chars.
+- Vendor keys MUST begin with a lowercase letter or a digit,
+  and can only contain lowercase letters (`a-z`),
+  digits (`0-9`), underscores (`_`), dashes (`-`), asterisks (`*`),
+  and forward slashes (`/`).
+- Vendor values have a maximum size of 256 chars.
+- Vendor values may only contain ASCII RFC0020 characters (i.e., the range `0x20` to `0x7E`) except comma `,` and `=`.
+- The keys and values used in the `elastic` entry must not contain the characters `:` and `;`.
```
Comment (axw) [inline docs/agents/distributed-tracing.md]:
```suggestion
- In addition to the above limitations, the keys and values used in the `elastic` entry must not contain the characters `:` and `;`.
```

just to make it clear that they can't contain `,`, `=`, ...

File: docs/agents/distributed-tracing.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+### Validation and length limits
+
+The [`tracestate`](https://www.w3.org/TR/trace-context/#tracestate-header)
+specification lists a number of validation rules.
+In addition to that,
+there are specific rules for the attributes under the `elastic` entry.
+
+Agents MUST implement these validation rules:
+
+- The `tracestate` field may contain a maximum of 32 entries.
```
Comment (axw) [inline docs/agents/distributed-tracing.md:43]: What do we do if there's already 32 entries, not including an `elastic` one?

**Review by axw — APPROVED** [review]

File: docs/agents/distributed-tracing.md (hunk 2 above)
Comment (axw) [reply docs/agents/distributed-tracing.md]: I'll go ahead and commit these suggestions so I can continue work on my PR without causing conflicts - please create a new PR to follow up as necessary.

File: docs/agents/distributed-tracing.md (hunk 3 above)
Comment (felixbarny) [reply docs/agents/distributed-tracing.md:43]: The spec says this: 

> Entries larger than 128 characters long SHOULD be removed first. Then entries SHOULD be removed starting from the end of tracestate.

Comment (felixbarny) [reply docs/agents/distributed-tracing.md:43]: but not including `elastic` would be the simplest strategy

**Pull request merged**